### ISSUE TITLE
Meta.source & Provenance

### ISSUE THREAD

**Issue by awatson1978** [AudaciousInquiry/fhir-saner#111]
Hello,
Quick question for the Intermediary/Hub implementations. Is there any guidance regarding how to model Provenance? We're going with `Meta.source` for the time being as we proxy MeasureReports from Cerner and relay them to LeidosWeb/NHSN.  But that assumes the MeasureReport is correctly formatted to begin with, and we're just relaying it on.  Also interested in the use cases around relaying APIs (such as from state or municipal governments, nursing home networks, etc); or if we have calculations and operations on the reports.  
Thanks!

**Comment by JohnMoehrke** [comment]
Yes @awatson1978 . The Basic Provenance IG explicitly covers intermediary. 
http://www.hl7.org/fhir/us/core/2019Sep/basic-provenance.html

**Comment by awatson1978** [comment]
Great.  I'll start implementing that in the Gateway/Proxy context.  

Next question...  the LeidosWeb HAPI server is overwriting `Meta.source`.  Is that expected behavior?

**Comment by awatson1978** [comment]
Also, this IG spells out how to do provenance for GET operations, but not PUSH operations?

**Comment by awatson1978** [comment]
Also, Keith asked me to open this ticket to track this item.  Let's not just close it right away?  Thanks.

**Comment by JohnMoehrke** [comment]
POST will be dependent on the capability of your POST use. 
Posting a bundle can just include the Provenance
Posting a single resource might be able to use the X header, if the server supports it.  http://build.fhir.org/provenance.html#header

So what more is needed? How is this need related to SANER? The issue definition seemed generic and not specific to SANER.

**Comment by awatson1978** [comment]
Yes, I agree X-Header is the way to go for the PUSH functionality, but it will need to be added to the SANER spec as an implementation detail.  

I'm concerned that X-Header doesn't get carried along with the payload though.  So, in the case of a municipal or state government acting as an intermediary and merging data into the reports from, say, nursing homes or schools, I'd think that we'd need a proper Bundle.  Which, to date, we haven't been testing in the SANER spec or in the Inferno testing.

**Comment by JohnMoehrke** [comment]
An intermediary that is not going to reflect the X-header, MUST that point forward us a transaction bundle that includes the Provenance detail. 

I am more worried that the x-header is simply not supported at all. So I would tend to say that SANER, if provenance is needed, MUST require the use of batch/transaction style of POST, rather than individual resource (create) post. Where that bundle contains the measure and the provenance of that measure. This is far more likely to be supported everywhere.
